packages/data-broker/src/services/weather/WeatherService.js

```diff
@@ -0,0 +1,267 @@
+import moment from 'moment';
+import { Service } from '../Service';
+import { ApiResultTranslator } from './ApiResultTranslator';
+import { DateSanitiser } from './DateSanitiser';
+
+export class WeatherService extends Service {
+  /**
+   * @param Object.<key: string, value: DatabaseModel> models
+   * @param WeatherApi api
+   */
+  constructor(models, api) {
+    super(models);
+    this.api = api;
+    this.dateSanitiser = new DateSanitiser();
+  }
+
+  /**
+   * Note, if no period specified will return current weather
+   * @inheritDoc
+   */
+  async pull(dataSources, type, options = {}) {
+    const { startDate, endDate } = options;
+
+    const resultFormat = this.getResultFormat(type);
+
+    const entities = await this.fetchEntitiesMatchingCodes(options.organisationUnitCodes);
+
+    const dataElementTypeDataSources = await this.extractDataSources(type, dataSources);
+
+    const forecastDataElementCodes = dataElementTypeDataSources
+      .filter(dataSource => dataSource.config.weatherForecastData === true)
+      .map(dataSource => dataSource.code);
+
+    const historicDataElementCodes = dataElementTypeDataSources
+      .filter(dataSource => !dataSource.config.weatherForecastData)
+      .map(dataSource => dataSource.code);
+
+    if (forecastDataElementCodes.length > 0 && historicDataElementCodes.length > 0) {
+      // no technical limitation, just not supported yet
+      throw new Error(
+        'Requesting both historic and forecast weather data in single call not supported',
+      );
+    }
+
+    if (forecastDataElementCodes.length > 0) {
+      const apiResultTranslator = new ApiResultTranslator(
+        entities,
+        resultFormat,
+        forecastDataElementCodes,
+      );
+
+      if (!this.isRequestForCurrentWeather(startDate, endDate)) {
+        // no technical reason, just not implemented yet
+        throw new Error('Date range not supported with forecast weather');
+      }
+
+      return this.getForecastWeatherForToday(entities, apiResultTranslator);
+    } else if (historicDataElementCodes.length > 0) {
+      const apiResultTranslator = new ApiResultTranslator(
+        entities,
+        resultFormat,
+        historicDataElementCodes,
+      );
+
+      if (this.isRequestForCurrentWeather(startDate, endDate)) {
+        throw new Error('Empty date range not supported with historic weather');
+      }
+
+      return this.getHistoricWeather(entities, startDate, endDate, apiResultTranslator);
+    }
+  }
+
+  /**
+   * @inheritDoc
+   */
+  async pullMetadata(dataSources, type) {
+    const dataElements = await this.models.dataSource.find({
+      type: 'dataElement',
+      service_type: 'weather',
+    });
+
+    const metadata = [];
+    for (const dataElement of dataElements) {
+      metadata.push({
+        code: dataElement.code,
+        name: dataElement.name,
+      });
+    }
+    return Promise.resolve(metadata);
```

Comment:
```suggestion
    return metadata;
```
I don't understand the need for the Promise.resolve, but I'm probably just missing something